src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java

```diff
@@ -30,22 +31,25 @@
     private final ExternalFileTypes externalFileTypes = mock(ExternalFileTypes.class);
     private final BibEntry entry = new BibEntry("article");
     private Path file;
-
+    private List<Path> dirs;
     @Rule public TemporaryFolder folder = new TemporaryFolder();
 
     @Before
     public void setUp() throws Exception {
         entry.setCiteKey("CiteKey");
         file = folder.newFile("CiteKey.pdf").toPath();
-        when(databaseContext.getFileDirectoriesAsPaths(any())).thenReturn(Collections.singletonList(folder.getRoot().toPath()));
+        dirs = Collections.singletonList(folder.getRoot().toPath());
+        when(databaseContext.getFileDirectoriesAsPaths(any())).thenReturn(dirs);
         when(externalFileTypes.getExternalFileTypeSelection()).thenReturn(new TreeSet<>(externalFileTypes.getDefaultExternalFileTypes()));
 
     }
 
     @Test
     public void test() {
         //Due to mocking the externalFileType class, the file extension will not be found
-        List<LinkedFile> expected = Collections.singletonList(new LinkedFile("", file.toString(), ""));
+
+        String relativeFilePath = FileUtil.shortenFileName(file, dirs).toString();
```

Comment:
Well, now the test uses the same logic as the method under test. Could you please hard code the expected value. It should be just `CiteKey.pdf`, right?